cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,1164 @@
+                       get_dataframe_buffer_begin(vertex_pair_buffer_p_q),
+                       edges,
+                       edges_last,
+                       });
+
+      thrust::for_each(handle.get_thrust_policy(),
+                       thrust::make_counting_iterator<edge_t>(0),
+                       thrust::make_counting_iterator<edge_t>(num_edge_exists),
+                       unroll_edge<vertex_t, edge_t, decltype(edges)>{
+                       raft::device_span<vertex_t>(num_triangles.data(), num_triangles.size()),
+                       get_dataframe_buffer_begin(vertex_pair_buffer_p_r),
+                       edges,
+                       edges_last,
+                       });     
+      
+      // Put edges with triangle count == 0 in the second partition
+      // FIXME: revisit all the 'stable_partition' and only used them
+      // when necessary otherwise simply call 'thrust::partition'
+      // Stable_parition is needed because we want to keep src and dst sorted
+      // so that we don't need to sort it again.
+      // FIXME: Create a rountine capturing L719:L763 as this block of code gets
+      // repeated
+      auto last_edge_with_triangles =
+        thrust::stable_partition(handle.get_thrust_policy(),
+                                 edges_to_num_triangles,
+                                 edges_to_num_triangles + num_edges,
+                                 [] __device__(auto edge_to_num_triangles) {
+                                   return thrust::get<1>(edge_to_num_triangles) > 0;
+                                 });
+
+      auto last_edge_with_triangles_idx = thrust::distance(edges_to_num_triangles, last_edge_with_triangles);
+      // rename the above it to last_edge_with_triangles
+
+      // Note: ensure 'edges_with_triangles' and 'cur_graph_view' have the same transpose flag
+      cugraph::edge_bucket_t<vertex_t, void, true, multi_gpu, true> edges_with_triangles(handle);
+
+      cugraph::edge_property_t<decltype(cur_graph_view), bool> edge_value_output(handle,
+                                                                                 cur_graph_view);
+
+      // rename the below to edges_with_triangles
+      edges_with_triangles.insert(edgelist_srcs.begin(),
+                                  edgelist_srcs.begin() + last_edge_with_triangles_idx,
+                                  edgelist_dsts.begin());                       
+
+      cugraph::transform_e(
+        handle,
+        cur_graph_view,
+        edges_with_triangles,
+        cugraph::edge_src_dummy_property_t{}.view(),
+        cugraph::edge_dst_dummy_property_t{}.view(),
+        cugraph::edge_dummy_property_t{}.view(),
+        [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+          return true;
+        },
+        edge_value_output.mutable_view(),
+        false);
+
+      cur_graph_view.attach_edge_mask(edge_value_output.view());
```

Comment:
I think edge_value_output (or better be renamed to edge_mask) should reside outside the while loop. This should have .view() object becomes invalid if the original object becomes out-of-scope.

Reply:
And you can update edge mask at the beginning of the loop once (right before applying detail::nbr_intersection for the edges to be removed). I don't see any need to apply this here.

Reply:
Right! the `edge_mask` is only useful for `nbr_intersection` so it doesn't make sense to apply it every time I remove edges. I can just apply it right before the `nbr_intersection` which in this case is in `case 1` when trying to unroll (p, q) edges.